### PR TITLE
feat: new ast-based helpers + fix has_args, has_decorators

### PR DESCRIPTION
**Pull request by Dario-DC**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #XXXXX

### CODE REVIEW

File: python/py_helpers.py [hunk 1]
```diff
@@ -424,6 +436,20 @@ def _find_conditions(tree):
 
         return [Node(test) for test in _find_conditions(self.tree)]
 
+    # Returs a Boolean indicating if node1 comes before node2
+    def is_ordered(self, node1, node2):
```
Comment (Dario-DC) [inline python/py_helpers.py:440]: Please, help me find a better name for this.

Comment (zairahira) [reply python/py_helpers.py:440]: The name is fine as it is serving the intended purpose 👍️

**Comment by zairahira** [comment]
Currently the `is_ordered()` method is checking for 2 statements. It would be better to allow variable number of statements to be checked.

```py
TypeError: Node.is_ordered() takes 3 positional arguments but 5 were given
```

**Comment by zairahira** [comment]
The method is also working perfect for non-consecutive statements. This could be mentioned in docs.

**Comment by zairahira** [comment]
The updated changes are working now for a variable number of arguments in `is_ordered()`👍️

**Review by zairahira — APPROVED** [review]

**Pull request merged**